pkg/virt-handler/non-root.go

```diff
@@ -8,21 +8,44 @@ import (
 	"strconv"
 	"strings"
 
+	k8sv1 "k8s.io/api/core/v1"
+
 	v1 "kubevirt.io/api/core/v1"
 	diskutils "kubevirt.io/kubevirt/pkg/ephemeral-disk-utils"
 	hostdisk "kubevirt.io/kubevirt/pkg/host-disk"
+	"kubevirt.io/kubevirt/pkg/util/types"
 	"kubevirt.io/kubevirt/pkg/virt-handler/isolation"
 	"kubevirt.io/kubevirt/pkg/virt-handler/selinux"
 )
 
-func changeOwnershipOfBlockDevices(vmiWithOnlyBlockPVCs *v1.VirtualMachineInstance, res isolation.IsolationResult) error {
-	for i := range vmiWithOnlyBlockPVCs.Spec.Volumes {
-		if volumeSource := &vmiWithOnlyBlockPVCs.Spec.Volumes[i].VolumeSource; volumeSource.PersistentVolumeClaim != nil {
-			devPath := filepath.Join(string(filepath.Separator), "dev", vmiWithOnlyBlockPVCs.Spec.Volumes[i].Name)
-			if err := diskutils.DefaultOwnershipManager.SetFileOwnership(filepath.Join(res.MountRoot(), devPath)); err != nil {
-				return err
-			}
+func changeOwnershipOfBlockDevices(vmi *v1.VirtualMachineInstance, res isolation.IsolationResult) error {
+	volumeModes := map[string]*k8sv1.PersistentVolumeMode{}
+	for _, volumeStatus := range vmi.Status.VolumeStatus {
+		if volumeStatus.PersistentVolumeClaimInfo != nil {
+			volumeModes[volumeStatus.Name] = volumeStatus.PersistentVolumeClaimInfo.VolumeMode
+		}
+	}
+
+	for i := range vmi.Spec.Volumes {
+		volume := vmi.Spec.Volumes[i]
+		if volume.VolumeSource.PersistentVolumeClaim == nil {
+			continue
+		}
+
+		volumeMode, exists := volumeModes[volume.Name]
+		if !exists {
+			return fmt.Errorf("missing volume status for volume %s", volume.Name)
 		}
+
+		if !types.IsPVCBlock(volumeMode) {
+			continue
```

Comment:
I would think that here we will need to change ownership of the volume mount path...
I thinking about a filesystem PV, not a block. Do we have a test where we boot a VM with a filesystem volume?

Reply:
Filesystem volumes are handled here https://github.com/kubevirt/kubevirt/blob/main/pkg/host-disk/host-disk.go#L227

Reply:
yep, that's right, thanks.